src/Builder/Stage.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder;
+
+use MongoDB\BSON\Document;
+use MongoDB\BSON\Serializable;
+use MongoDB\Builder\Stage\MatchStage;
+use MongoDB\Builder\Type\FieldQueryInterface;
+use MongoDB\Builder\Type\QueryInterface;
+use stdClass;
+
+enum Stage
+{
+    use Stage\FactoryTrait {
+        match as private generatedMatch;
+    }
+
+    /**
+     * Filters the document stream to allow only matching documents to pass unmodified into the next pipeline stage. $match uses standard MongoDB queries. For each input document, outputs either one document (a match) or zero documents (no match).
+     *
+     * @see https://www.mongodb.com/docs/manual/reference/operator/aggregation/match/
+     *
+     * @param Document|FieldQueryInterface|QueryInterface|Serializable|array|stdClass $queries
+     */
+    public static function match(FieldQueryInterface|QueryInterface|Serializable|array|bool|float|int|stdClass|string|null ...$queries): MatchStage
+    {
+        // Override the generated method to allow variadic arguments
```

Comment:
Is there a reason the generated method doesn't accept var args on its own?

Reply:
It's something I can work on later. Query can be passed to operators in all forms:
- As single param: `$elemMatch`, `$match`. Variadic is great for them.
- As a list of query: `$and`, `$or`, `$nor`, Variadic can't be used.
- With other params: `$geoNear`, `$graphLookup`. Variadic can't be used.
- `$not` accept a single fieldQuery object without field name. So not variadic.